### ISSUE TITLE
CORS Content-Type "bypass" 2

### ISSUE THREAD

**Issue by annevk** [whatwg/fetch#838]
@MattMenke2 made me think of another "bypass", though it would only affect naïve parsers:

* `text/plain;,application/json`
* `text/plain;charset=utf-8,application/json`

These would only be successful with a server that splits on `,` and uses the last value, which isn't compliant per HTTP. An easy fix would be to ban 0x2C `,`.

I thought I'd raise it here so it can be considered. I'm sorry for all the recent changes to this and that not everything was considered at once. Hopefully the bugs and tests help mitigate the churn annoyance.

cc @yutakahirano @youennf

**Comment by annevk** [comment]
@whatwg/security thoughts? Shall I make patches for banning `,`?

**Comment by antosart** [comment]
I like the idea of banning `,`, meaning we would always issue a preflight in that case.

**Comment by annevk** [comment]
Yeah, that seems by far the most straightforward. I suppose the alternative would be something like this:

In https://fetch.spec.whatwg.org/#cors-safelisted-request-header under `content-type`, replace step 2-4 with:

1. Let _splitValues_ be the result of getting, decoding, and splitting `Content-Type` from « _header_ ». (This syntax is somewhat confusing maybe, but turns _header_ back into a header list, which "getting, decoding, and splitting" needs. An alternative might be to make "decode and split" its own algorithm separate from getting.)
2. If _splitValues_ is null, then return failure.
3. For each _splitValue_ of _splitValues_:
   1. Let _mimeType_ be the result of parsing _splitValue_.
   2. If _mimeType_ is failure, then return false.
   3. If mimeType’s essence is not "application/x-www-form-urlencoded", "multipart/form-data", or "text/plain", then return false.
